### ISSUE TITLE
Attempt to Compare Number with Boolean Error (WOTLK Classic)

### ISSUE THREAD

**Issue by salt-mountain** [Jaliborc/Bagnon#1848]
**Which software were you running?**
 - Addon version name: Bagnon-10.2.12
 - Client used: WOTLK Classic 3.4.3.52237

**Have you read the changelog?**
Yes

**Please describe the bug.**
Trying to open the bags simply errors out and doesn't open the bag frame.

**Please describe how to reproduce it.**
1. Login to your Character on WOTLK
2. Try to open your bags by hitting either the "B" button or clicking the Bag icon in the Micro Menu.
3. See error

**Any Screenshots?**
If applicable, please add screenshots to help explain your problem.

**Error Logs are Important!**
Type `/console scriptErrors 1` in the chat and reload the game. If an error window appears when reproducing the bug, please write here the first message (labelled 1/Many) that appeared. **:warning: By default, the last error message is shown, not the first!** You need to navigate back to the first error.
```
14x BagBrother/core/classes/itemGroup.lua:124: attempt to compare number with boolean
[string "@BagBrother/core/classes/itemGroup.lua"]:124: in function `Layout'
[string "@BagBrother/core/classes/itemGroup.lua"]:32: in function `Update'
[string "@BagBrother/core/classes/itemGroup.lua"]:26: in function <BagBrother/core/classes/itemGroup.lua:12>
[string "=(tail call)"]: ?
[string "@Bagnon/src/frame.lua"]:28: in function `New'
[string "@BagBrother/core/api/frames.lua"]:83: in function `New'
[string "@BagBrother/core/api/frames.lua"]:28: in function <BagBrother/core/api/frames.lua:27>
[string "=(tail call)"]: ?
[string "@BagBrother/core/features/autoDisplay.lua"]:83: in function `OpenAllBags'
[string "@Interface_TBC/FrameXML/MerchantFrame.lua"]:88: in function <Interface_TBC/FrameXML/MerchantFrame.lua:87>
[string "=[C]"]: in function `Show'
[string "@Interface_Wrath/FrameXML/UIParent.lua"]:2282: in function `SetUIPanel'
[string "@Interface_Wrath/FrameXML/UIParent.lua"]:2127: in function `ShowUIPanel'
[string "@Interface_Wrath/FrameXML/UIParent.lua"]:1995: in function <Interface_Wrath/FrameXML/UIParent.lua:1991>
[string "=[C]"]: in function `SetAttribute'
[string "@Interface_Wrath/FrameXML/UIParent.lua"]:2806: in function `ShowUIPanel'
[string "@Interface_TBC/FrameXML/MerchantFrame.lua"]:45: in function `showFunc'
[string "@FrameXML/PlayerInteractionFrameManager.lua"]:183: in function <FrameXML/PlayerInteractionFrameManager.lua:172>
[string "=(tail call)"]: ?
[string "@FrameXML/PlayerInteractionFrameManager.lua"]:215: in function <FrameXML/PlayerInteractionFrameManager.lua:212>

Locals:
self = Frame {
 0 = <userdata>
 buttons = <table> {
 }
 order = <table> {
 }
 frame = BagnonInventory1 {
 }
 bags = <table> {
 }
}
profile = <table> {
 brokerObject = "BagnonLauncher"
 columns = 8
 hiddenBags = <table> {
 }
 color = <table> {
 }
 lockedSlots = <table> {
 }
 hiddenRules = <table> {
 }
 rules = <table> {
 }
 bagBreak = false
 x = -54.941895
 showBags = true
 borderColor = <table> {
 }
 reverseSlots = false
 point = "BOTTOMRIGHT"
 y = 24.944738
 reverseBags = false
}
columns = 8
scale = 1
size = 39
revBags = false
revSlots = false
x = 4
y = 2
group = 0
(for index) = 2
(for limit) = 6
(for step) = 1
k = 2
bag = 1
slots = <table> {
 1 = BagnonContainerItem21 {
 }
 2 = BagnonContainerItem22 {
 }
 3 = BagnonContainerItem23 {
 }
 4 = BagnonContainerItem24 {
 }
 5 = BagnonContainerItem25 {
 }
 6 = BagnonContainerItem26 {
 }
 7 = BagnonContainerItem27 {
 }
 8 = BagnonContainerItem28 {
 }
 9 = BagnonContainerItem29 {
 }
 10 = BagnonContainerItem30 {
 }
 11 = BagnonContainerItem31 {
 }
 12 = BagnonContainerItem32 {
 }
 13 = BagnonContainerItem33 {
 }
 14 = BagnonContainerItem34 {
 }
 15 = BagnonContainerItem35 {
 }
 16 = BagnonContainerItem36 {
 }
 17 = BagnonContainerItem37 {
 }
 18 = BagnonContainerItem38 {
 }
 19 = BagnonContainerItem39 {
 }
 20 = BagnonContainerItem40 {
 }
 21 = BagnonContainerItem41 {
 }
 22 = BagnonContainerItem42 {
 }
}
family = 0
(*temporary) = false
(*temporary) = 1
(*temporary) = 0
(*temporary) = 2
(*temporary) = 0
(*temporary) = 0
(*temporary) = 0
(*temporary) = 0
(*temporary) = 0
(*temporary) = "owner"
(*temporary) = nil
(*temporary) = nil
(*temporary) = <table> {
 PlaceItemGroup = <function> defined @Bagnon/src/frame.lua:252
 GetOwner = <function> defined @BagBrother/core/classes/frameBase.lua:159
 __base = <table> {
 }
 Pools = <table> {
 }
 IsBagGroupShown = <function> defined @Bagnon/src/frame.lua:262
 IsShowingBag = <function> defined @BagBrother/core/classes/frameBase.lua:97
 BagGroup = <table> {
 }
 SetOwner = <function> defined @BagBrother/core/classes/frameBase.lua:154
 HasOwnerSelector = <function> defined @Bagnon/src/frame.lua:217
 Layout = <function> defined @Bagnon/src/frame.lua:104
 FindRules = <function> defined @BagBrother/core/classes/frameBase.lua:88
 IsCached = <function> defined @BagBrother/core/classes/frameBase.lua:142
 PlaceBrokerCarrousel = <function> defined @Bagnon/src/frame.lua:300
 PlaceCurrencies = <function> defined @Bagnon/src/frame.lua:282
 PlaceMoney = <function> defined @Bagnon/src/frame.lua:269
 __super = <table> {
 }
 __frames = <table> {
 }
 __type = "Frame"
 __count = 0
 GetFrameID = <function> defined @BagBrother/core/classes/frameBase.lua:163
 IsShowingQuality = <function> defined @BagBrother/core/classes/frameBase.lua:114
 PlaceMenuButtons = <function> defined @Bagnon/src/frame.lua:133
 HasSearchToggle = <function> defined @Bagnon/src/frame.lua:221
 PlaceOptionsToggle = <function> defined @Bagnon/src/frame.lua:174
 HasCurrencies = <function> defined @Bagnon/src/frame.lua:325
 UpdateBackdrop = <function> defined @Bagnon/src/frame.lua:66
 GetBaseProfile = <function> defined @BagBrother/core/classes/frameBase.lua:150
 RegisterSignals = <function> defined @Bagnon/src/frame.lua:47
 OpenSound = 862
 PlaceSearchBar = <function> defined @Bagnon/src/frame.lua:157
 Update = <function> defined @Bagnon/src/frame.lua:59
 SortItems = <function> defined @BagBrother/core/classes/frameBase.lua:118
 SetPosition = <function> defined @BagBrother/core/classes/frameBase.lua:72
 BrokerSpacing = 2
 IsShowingItem = <function> defined @BagBrother/core/classes/frameBase.lua:101
 HasSortButton = <function> defined @Bagnon/s
 ```

**Comment by salt-mountain** [comment]
Switching back to `Bagnon-10.2.9` seems to allow me to open my bags again, but presents a different LUA error.
```
1x BagBrother/core/features/updateDetection.lua:19: attempt to call method 'match' (a nil value)
[string "@BagBrother/core/features/updateDetection.lua"]:19: in function <BagBrother/core/features/updateDetection.lua:18>
[string "@BagBrother/core/features/updateDetection.lua"]:33: in function `?'
[string "@BagBrother/libs/WildAddon-1.0-2/WildAddon-1.0.lua"]:20: in function <...dOns/BagBrother/libs/WildAddon-1.0/WildAddon-1.0.lua:18>
[string "@BagBrother/libs/WildAddon-1.0-2/WildAddon-1.0.lua"]:56: in function <...dOns/BagBrother/libs/WildAddon-1.0/WildAddon-1.0.lua:53>
[string "=[C]"]: ?
[string "@Attune/Libs/CallbackHandler-1.0-8/CallbackHandler-1.0.lua"]:19: in function <...une/Libs/CallbackHandler-1.0/CallbackHandler-1.0.lua:15>
[string "@Attune/Libs/CallbackHandler-1.0-8/CallbackHandler-1.0.lua"]:54: in function `Fire'
[string "@Attune/Libs/AceEvent-3.0-4/AceEvent-3.0.lua"]:120: in function <...aceAttune/Libs/AceEvent-3.0/AceEvent-3.0.lua:119>
[string "=[C]"]: in function `LoadAddOn'
[string "@Interface_Wrath/FrameXML/UIParent.lua"]:398: in function `UIParentLoadAddOn'
[string "@Interface_Wrath/FrameXML/UIParent.lua"]:489: in function `TimeManager_LoadUI'
[string "@Interface_Wrath/FrameXML/UIParent.lua"]:846: in function <Interface_Wrath/FrameXML/UIParent.lua:815>

Locals:
version = <table> {
}
(*temporary) = nil
(*temporary) = <table> {
}
(*temporary) = "(%d+)%.*(%d*)%.*(%d*)"
(*temporary) = "attempt to call method 'match' (a nil value)"
```

**Comment by salt-mountain** [comment]
Seemingly fixed / not occurring in `Bagnon-10.2.13`
